Pagination/Configuration/PaginateFinderConfiguration.php

```diff
@@ -14,29 +14,51 @@ class PaginateFinderConfiguration extends FinderConfiguration
     protected $limit = null;
 
     /**
-     * @param Request $request
+     * @param FinderConfiguration $configuration
+     * @param null|array          $order
+     * @param null|int            $skip
+     * @param null|int            $limit
      *
      * @return PaginateFinderConfiguration
      */
-    public static function generateFromRequest(Request $request)
+    public static function generatePaginateFromVariable(FinderConfiguration  $configuration, $order =null, $skip = null, $limit = null)
     {
         $finderConfig = new PaginateFinderConfiguration();
-        $columns = $request->get('columns');
-        if(PaginateFinderConfiguration::isArrayOrNull($columns)){
-            $finderConfig->setColumns($columns);
-        }
-        $finderConfig->setSearch($request->get('search'));
-        $finderConfig->setLimit($request->get('limit'));
-        $order = $request->get('order');
+
+        $finderConfig->setColumns($configuration->getColumns());
+        $finderConfig->setSearch($configuration->getSearch());
+        $finderConfig->setDescriptionEntity($configuration->getDescriptionEntity());
```

Comment:
with new self() in generateFromVariable, you will not need new PaginateFinderConfiguration() and not these three lines and could set extra parameters directly in $configuration.